### DISCUSSION TITLE
When I use from setColumns, other columns lose them format

### DISCUSSION THREAD

**Discussion by haajit** [serenity-is/Serenity#6485] [Help] [answered]
when I use setColumns for hide some columns, other columns lose them format

            let columns = this.getColumns();

            if (this.toolbar.element.find('.fa-chain-broken').length > 0) {
                columns = columns.filter(f => f.field != 'Name');
                columns = columns.filter(f => f.field != 'NameEn');
                columns = columns.filter(f => f.field != 'Code');
            }
            else {
                columns = columns.filter(f => f.field != 'MsName');
                columns = columns.filter(f => f.field != 'MsCode');
            }
            this.slickGrid.setColumns(columns);
Original format:
![image](https://user-images.githubusercontent.com/9511137/183247478-6c557a77-80fe-4c90-b4e7-6e9566c80ad8.png)

Lost format:
![image](https://user-images.githubusercontent.com/9511137/183247405-3d124142-f6e4-4f6e-bc5c-a563de82c8e4.png)

**Comment by stixoffire** [comment]
I believe this is a slickGrid issue you need to rebuild your formatting .  

Basically you need to re-render to preserve the formatting .

Try this after you set columns 
this.slickGrid.render().

**Reply by haajit** [reply]
> I believe this is a slickGrid issue you need to rebuild your formatting .
> 
> Basically you need to re-render to preserve the formatting .
> 
> Try this after you set columns this.slickGrid.render().

Thanks for your answer,
Unfortunately it doesn't work for me

**Comment by haajit** [comment, marked as answer]
I added this code  

            columns.forEach(x => {
                x.formatter = Serenity.SlickHelper.convertToFormatter(x.format);
            });
befor  this.slickGrid.setColumns(columns);
and solved my problem
